core/lib/protobuf_config/src/proto/config/prover.proto

```diff
@@ -108,4 +108,5 @@ message ProofDataHandler {
   optional uint32 http_port = 1; // required; u16
   optional uint32 proof_generation_timeout_in_secs = 2; // required; s
   optional bool tee_support = 3; // required
+  optional uint64 first_tee_processed_batch = 4; // required
```

Comment:
How do you set the default of `first_tee_processed_batch` to `0` in a `*.proto` file? To be honest, I don’t really understand how these proto files work in our case – I treat it as boilerplate that I need to revisit whenever modifying configuration based on environment variables. For example, we have:
```
optional uint32 http_port = 1; // required; u16
```
Notice `uint32` vs `u16`? Is this just for documentation to indicate that `http_port` is actually represented as `u16` in the code? If so, it can easily get out of sync (unless it's checked in CI), and I wouldn't rely on it anyway.

Reply:
Yes, it's just a documentation thing. It's sort of checked in CI by [parsing sample file-based configs](https://github.com/matter-labs/zksync-era/blob/b29be7d9a8c664beac5d8384548db54de0ba882f/core/lib/protobuf_config/src/tests.rs#L61). You do not set a default in the proto file, you set it in the glue code, i.e., in a conversion from the Protobuf type to the domain type.

My moderately strong opinion is that the language / tooling constraints are not a reason to write lower-quality code; i.e., in this case lack of defaults / constrained Protobuf type system are not good reasons to degrade UX or desync config parsing logic compared to env-based config. (As a side note, AFAIU, we plan to eventually remove either Protobuf-based or serde-based config, but haven't 100% decided which one yet 🙃)